data/helpers.d/utils

```diff
@@ -492,6 +492,12 @@ ynh_check_app_version_changed () {
 # example: ynh_compare_package_version --comparison gt --version 2.3.2~ynh1
 # This example will check if the installed version is greater than (gt) the version 2.3.2~ynh1
 #
+# Generally you might probably use it as follow in the upgrade script
+#
+# if ynh_compare_package_version --comparaison gt --version 2.3.2~ynh1; then
+#     # Do something that is needed for the package version older than 2.3.2~ynh1
```

Comment:
Huh I may have not fully understand in which way it works...
If the current version is greater than 2.3.2\~ynh1 (Yet I don't see any situation I would do that, rather the opposite), you say it does something for package older than 2.3.2\~ynh1...

My brain is in knots...

My new package version is 2.0
I want to migrate from 1.0 to 2.0
if ynh_compare_package_version --comparison lt --version 2.0~ynh1
then
  I do my migration
fi
Because my current version is 1.0, so my version is lower than 2.0

Is that it ?

Reply:
Oh yes it's the opposite.

Reply:
Maybe a clear explanation then

Reply:
A French explanation from https://debian-handbook.info/browse/fr-FR/stable/sect.manipulating-packages-with-dpkg.html

dpkg étant le programme de référence pour manipuler les paquets Debian, il fournit également l'implémentation de référence de la logique de comparaison des numéros de versions. C'est pourquoi il dispose d'une option --compare-versions utilisable par des programmes externes (et notamment les scripts de configuration exécutés par dpkg lui-même). Cette option requiert trois paramètres : un numéro de version, un opérateur de comparaison et un second numéro de version. Les différents opérateurs possibles sont lt (strictement plus petit que — lower than), le (plus petit ou égal à — lower or equal), eq (égal à — equal), ne (différent de — not equal), ge (plus grand ou égal à — greater or equal) et gt (strictement plus grand que — greater than). Si la comparaison est avérée, dpkg renvoie le code de retour 0 (succès) ; sinon il renvoie une valeur non nulle (indiquant un échec).

And the most important is the last line:
Return 0 if it's true
Return not 0 if it's false....

Reply:
@yalh76 you do realize that I'm not asking an explanation for me.
My concern is that when a packager will use that helper, the expected behavior was clear.

Considering that I can read the code and yet I ask if I correctly understood how the helper works. I expect packagers to ask and not be sure about this helper.

So, if when reading the helper, I ask for explanation, I think this is not clear enough.

Reply:
Yes I know that is not clear, and you don't need it.

I just though that adding some french explanations for dpkg --compare-versions could help to improve the documentation in the helper....

Reply:
dpkg --compare-versions is implemented in wekan_ynh... and it took me several reads to understand it

Reply:
the english translation if it could help.. or not:
https://debian-handbook.info/browse/stable/sect.manipulating-packages-with-dpkg.html

Reply:
Soo, as you are better than me to write the documentation, what would you want that write.

Personally I don't think that it would be good to write in French as it's also dedicated to the English packager (or maybe I didn't understand what you mean).

Reply:
well... as I'm not in the core group, just proposing adjustments in https://github.com/YunoHost/yunohost/pull/934

I think too that french is bad...